File: Espoir/Entities/EspoirContext.cs
using Espoir.Helpers;

namespace Espoir.Entities
{
    internal class EspoirContext
    {
        const int MAX_PLAYERS = 100;

        public Player MainCharacter { get; set; }

        public IList<Player> Players { get; set; }
        
        public Repeater TenMinuteTask { get; private set; }

        public EspoirContext()
        {
            this.MainCharacter = new Player();
            this.Players = new List<Player> { this.MainCharacter };
            for (int i=1; i<MAX_PLAYERS; i++)
            {
                this.Players.Add(NonPlayablePlayer.Generate(i));
            }

            this.TenMinuteTask = new Repeater(this.TenMinuteTick);
        }

        public void StartGame()
        {
            this.TenMinuteTask.Start();
        }

        public void FinishGame()
        {
            this.TenMinuteTask.Stop();
        }

        public IDictionary<CardType, int> GetCardFrequency()
        {
            return this.Players
                .SelectMany(p => p.Cards)
                .GroupBy(x => x)
                .ToDictionary(x => x.Key, x => x.Count());
        }

        public async Task TenMinuteTick(CancellationToken token)
        {
            while (true)
            {
                var tenMinutes = 60 * 10 * 1000;
                await Task.Delay(tenMinutes, token);
                if (!token.IsCancellationRequested)
                {
                    this.MainCharacter.Debt *= new decimal(1.0150);
                }
                else
                {
                    break;
                }
            }
        }
    }
}

File: renpy-version/entities/espoir_context.py
from collections import Counter
from entities.non_playable_player import NonPlayablePlayer
from entities.player import Player

MAX_PLAYERS = 100

class EspoirContext:
    def __init__(self):
        self.MainCharacter = Player()
        self.Players = [self.MainCharacter]
        for i in range(1, MAX_PLAYERS):
            self.Players.append(NonPlayablePlayer(i))
        
        # TODO: 10-min task

    def startGame():
        pass

    def finishGame():
        pass

    def getCardFrequency(self):
        return Counter(self.Cards)